### ISSUE TITLE
Inconsistence in istioctl upgrade on 1.12 version

### ISSUE THREAD

**Issue by cnvergence** [istio/istio#37908]
### Bug Description

Installing default 1.12.1 Istio on the cluster with `istioctl install --set profile=default -y `
and upgrading it to 1.12.2 and then to 1.12.3 with `istioctl upgrade --set profile=default -y`
will result in the following error:
```
Control Plane - istio-ingressgateway pod - istio-ingressgateway-8c48d875-zkztr - version: 1.12.2
Control Plane - istiod pod - istiod-58d79b7bff-7qfrs - version: 1.12.2

Upgrade version check passed: 1.12.2 -> 1.12.3.

Upgrade check: Warning!!! The following IOPS will be changed as part of upgrade. Please double check they are correct:
spec:
  installPackagePath: /var/folders/w9/y0g091cn7fb7shnmqw84vp_m0000gn/T/istio-install-packages/istio-1.12.1/manifests
    -> <empty> (REMOVED)
  values:
    pilot:
      env:
        ENABLE_LEGACY_FSGROUP_INJECTION: <empty> -> false (ADDED)

2022-03-14T14:32:58.144294Z	info	Error: failed to apply the Istio Control Plane specs. Error: errors occurred during operation: creating default tag would conflict:
Error [IST0139] (MutatingWebhookConfiguration istio-sidecar-injector) Webhook overlaps with others: [istio-revision-tag-default/namespace.sidecar-injector.istio.io]. This may cause injection to occur twice.
Error [IST0139] (MutatingWebhookConfiguration istio-sidecar-injector) Webhook overlaps with others: [istio-revision-tag-default/object.sidecar-injector.istio.io]. This may cause injection to occur twice.
Error [IST0139] (MutatingWebhookConfiguration istio-sidecar-injector) Webhook overlaps with others: [istio-revision-tag-default/rev.namespace.sidecar-injector.istio.io]. This may cause injection to occur twice.
Error [IST0139] (MutatingWebhookConfiguration istio-sidecar-injector) Webhook overlaps with others: [istio-revision-tag-default/rev.object.sidecar-injector.istio.io]. This may cause injection to occur twice.

Error: failed to apply the Istio Control Plane specs. Error: errors occurred during operation: creating default tag would conflict:
Error [IST0139] (MutatingWebhookConfiguration istio-sidecar-injector) Webhook overlaps with others: [istio-revision-tag-default/namespace.sidecar-injector.istio.io]. This may cause injection to occur twice.
Error [IST0139] (MutatingWebhookConfiguration istio-sidecar-injector) Webhook overlaps with others: [istio-revision-tag-default/object.sidecar-injector.istio.io]. This may cause injection to occur twice.
Error [IST0139] (MutatingWebhookConfiguration istio-sidecar-injector) Webhook overlaps with others: [istio-revision-tag-default/rev.namespace.sidecar-injector.istio.io]. This may cause injection to occur twice.
Error [IST0139] (MutatingWebhookConfiguration istio-sidecar-injector) Webhook overlaps with others: [istio-revision-tag-default/rev.object.sidecar-injector.istio.io]. This may cause injection to occur twice.
```

### Version

```prose
client version: 1.12.3
control plane version: 1.12.2
data plane version: 1.12.2 (1 proxies)
```

### Additional Information

This is probably related to the fact that the upgrade command does not check if the default revision exists on the cluster, as the install command does.
The newer version should not have this issue, as there istioctl upgrade is an alias to the istioctl install.

The question is if it should be fixed in the release-1.12 or not

**Comment by litong01** [comment]
@cnvergence were you using `upgrade` command going from 1.12.1 to 1.12.2? would like to have your confirmation so that I can try this to recreate the issue. This webhook thing has been a headache lately.

**Comment by cnvergence** [comment]
yes, this is only related to upgrade command

**Comment by litong01** [comment]
also tried from 1.12.3 to 1.12.4 to 1.12.5 same error.

**Comment by cnvergence** [comment]
that's correct, I believe the revision tag logic from install, should be also included in the upgrade on 1.12
